lib/PreviewProvider.js

```diff
@@ -277,15 +221,27 @@ PreviewProvider.prototype = {
   }),
 
   /**
+   * Find the metadata for each link in the database
+   */
+  _asyncFindItemsInDB: Task.async(function*(links) {
+    let cacheKeyArray = [];
+    links.forEach(link => cacheKeyArray.push(link.cache_key));
+    let items = yield this._metadataStore.asyncGetMetadataByCacheKey(cacheKeyArray);
+    return items;
+  }),
+
+  /**
    * Request links from embedly, optionally filtering out known links
    */
-  _asyncSaveLinks: Task.async(function*(processedLinks, event, updateAccessTime = true) {
+  _asyncSaveLinks: Task.async(function*(processedLinks, event) {
+    let DBItems = yield this._asyncFindItemsInDB(processedLinks);
+    let setItems = new Set();
```

Comment:
I think this is storing the links for which we already have data in the db, so we should call it something like

existingLinks